### ISSUE TITLE
Marshalling of UDF return value fails in debug builds under 64-bit Excel

### ISSUE THREAD

**Issue by chelh** [Excel-DNA/ExcelDna#96]
Hi Govert, and thanks for making and maintaining Excel-DNA!

I'm building Excel-DNA myself from the source code, the latest version straight off GitHub. When I build Excel-DNA in debug configuration, then use it in an add-in under 64-bit Excel, all of the add-in's UDFs return `#NUM!` or 0. This is reproducible even using a very simple test add-in:

```xml
<DnaLibrary Name="ExcelDnaTest Add-In" RuntimeVersion="v4.0" Language="CS">
  <![CDATA[
  public static class Test {
      public static object Test1(string arg) {
          return arg;
      }
  }
  ]]>
</DnaLibrary>
```

A similar bug, which I've been unable to reproduce in a test add-in, involves all arguments being marshalled over as `#VALUE!` as well. Here's a screenshot from trying to debug this with my larger (proprietary) add-in:

![debugging XlCustomMarshal.cs](https://cloud.githubusercontent.com/assets/14240661/20428644/13cb18a2-ad3f-11e6-9077-dc16f1c3fc56.png)

I've attached my own build of Excel-DNA, in case you cannot reproduce:

[ExcelDna_AMD64.zip](https://github.com/Excel-DNA/ExcelDna/files/599800/ExcelDna_AMD64.zip)

Thanks for any help you can provide. This isn't super high-priority since it doesn't happen in release builds, but it does impede my debugging, so it would be nice to get fixed eventually!

**Comment by govert** [comment]
There's a `DEBUG` flag that changes the marshaling configuration under `DEBUG` to be the version compatible with Excel versions pre-2007. I don't think this marshaling implementation work right for 64-bit Excel. You can disable this flag by editing this line: https://github.com/Excel-DNA/ExcelDna/blob/master/Source/ExcelDna.Loader/XlParameterInfo.cs#L94. Then you'll be running the same code under debug and release, which should sort out your problem.

**Comment by govert** [comment]
The errant DEBUG flag is fixed in commit 004c5ff24567a8b5a36866ff4498d8ebf011fd35
